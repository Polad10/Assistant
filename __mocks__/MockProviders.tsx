import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
