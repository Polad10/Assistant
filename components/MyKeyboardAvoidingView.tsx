import { ReactNode, useCallback, useEffect, useRef } from 'react'
import { Keyboard, KeyboardAvoidingView, ScrollView } from 'react-native'

type Props = {
  children: ReactNode
  focusedInputIndex: number
}

export default function MyKeyboardAvoidingView(props: Props) {
  const scrollRef = useRef<ScrollView>(null)

  const scrollToInput = useCallback(() => {
    scrollRef.current?.scrollTo({ x: 0, y: 120 * props.focusedInputIndex })
  }, [props.focusedInputIndex])

  useEffect(() => {
    const listener = Keyboard.addListener('keyboardWillShow', scrollToInput)

    return () => {
      listener.remove()
    }
  }, [scrollToInput])

  return (
    <KeyboardAvoidingView style={{ flex: 1 }} behavior='padding' keyboardVerticalOffset={150}>
      <ScrollView contentContainerStyle={{ flexGrow: 1 }} ref={scrollRef}>
        {props.children}
      </ScrollView>
    </KeyboardAvoidingView>
  )
}
