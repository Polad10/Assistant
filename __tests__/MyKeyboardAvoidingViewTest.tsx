import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import MyKeyboardAvoidingView from '../components/MyKeyboardAvoidingView'

describe('MyKeyboardAvoidingView', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <MyKeyboardAvoidingView />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
