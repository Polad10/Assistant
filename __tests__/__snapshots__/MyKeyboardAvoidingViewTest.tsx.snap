// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyKeyboardAvoidingView renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={50}
  >
    <View />
  </RCTScrollView>
</View>
`;
